src/HttpClient.php

```diff
@@ -21,7 +21,8 @@
      *
      * @return ResponseInterface
      *
-     * @throws Exception
+     * @throws Exception  If an error happens during processing the request.
+     * @throws \Exception If processing the request is impossible (eg. bad configuration).
```

Comment:
Can we use this exception naming instead:

```php
@throws \Http\Exception
@throws \Exception
```

Currently it's feel a bit weird and i'm afraid of misunderstanding from user, if it's read like that